src/Cli.Tests/EndToEndTests.cs

```diff
@@ -131,6 +131,44 @@ public void TestInitializingRestAndGraphQLGlobalSettings()
         Assert.IsTrue(runtimeConfig.Runtime.GraphQL?.Enabled);
     }
 
+    /// <summary>
+    /// Test to validate the successful generation of config file with the --graphql.nested-insert.enabled option of the init command.
+    /// </summary>
+    /// <param name="isNestedInsertsEnabled">Value for the nested insert enabled flag in the init command.</param>
+    /// <param name="expectedValueForNestedInsertEnabledFlag"> Expected value for the nested insert enabled flag in the config file.</param>
+    [DataTestMethod]
+    [DataRow(CliBool.True, true, DisplayName = "Nested Insert operation is enabled the config file by specifying '--graphql.nested-insert.enabled true'")]
+    [DataRow(CliBool.False, false, DisplayName = "Nested Insert operation is disabled the config file by specifying '--graphql.nested-insert.enabled false'")]
+    [DataRow(null, false, DisplayName = " '--graphql.nested-insert' option is not used in the init command. When not enabled explicitly, the nested insert operation will be disabled.")]
+    public void TestEnablingNestedInsertOperation(CliBool? isNestedInsertsEnabled, bool expectedValueForNestedInsertEnabledFlag)
+    {
+        List<string> args = new() { "init", "-c", TEST_RUNTIME_CONFIG_FILE, "--connection-string", SAMPLE_TEST_CONN_STRING, "--database-type", "mssql" };
+
+        if (isNestedInsertsEnabled is not null)
+        {
+            args.Add("--graphql.nested-insert.enabled");
+            args.Add(isNestedInsertsEnabled.ToString()!);
+        }
+
+        Program.Execute(args.ToArray(), _cliLogger!, _fileSystem!, _runtimeConfigLoader!);
+
+        Assert.IsTrue(_runtimeConfigLoader!.TryLoadConfig(
+            TEST_RUNTIME_CONFIG_FILE,
+            out RuntimeConfig? runtimeConfig,
+            replaceEnvVar: true));
+
+        SqlConnectionStringBuilder builder = new(runtimeConfig.DataSource.ConnectionString);
+        Assert.AreEqual(ProductInfo.GetDataApiBuilderApplicationName(), builder.ApplicationName);
+
+        Assert.IsNotNull(runtimeConfig);
+        Assert.AreEqual(DatabaseType.MSSQL, runtimeConfig.DataSource.DatabaseType);
+        Assert.IsNotNull(runtimeConfig.Runtime);
+        Assert.IsNotNull(runtimeConfig.Runtime.GraphQL);
+        Assert.IsNotNull(runtimeConfig.Runtime.GraphQL.NestedMutationOptions);
```

Comment:
in case the option is not specified i.e. `isNestedInsertedEnabled` is null, there should be no change to the config. We should not be writing out the nested-mutations section at all. The config size will not increase as a result when people upgrade

Reply:
Writing out all the fields has been the approach we've adopted for other fields, and this is in line with that.
Some examples of this include: 
1. Writing out the `path` and `request-body-strict` fields even if `rest` is disabled
2. Likewise, `path`, `allow-introspection` fields for `graphQL`
3.  In the entities section, we again write out fields like `path`, `singular`, `plural` names, `source.fields`, `target.fields` of relationship (when fields are not specified),  `rest.methods`, `graphql.operations` and more. 

Also, writing out the fields has the following advantages,
1. Users get to know the presence of this new section in the config file. They might not always closely follow the latest changes to the documentation. 
2. Avoids any ambiguity on the default behavior when the option is not used. Even though, we would document the default behavior, they might not be caught with the latest additions to the docs.


I've tested to the changes with an old config file (without this new section) that it deserializes correctly and REST, GraphQL API requests continue to work fine. So, this let's be backwards compatible.

Reply:
are there tests for this: "I've tested the changes with an old config file (without the nested mutation fields) that it deserializes correctly and REST, GraphQL API requests work fine." 
When someone doesn't use the CLI, they won't have the new fields output to their config file. Need to test that excluding the new fields from the config or setting them to null has no negative effects.

Reply:
I noticed that we don't have any test that validates the backward compatibility aspect, specifically an older version of config file working successfully with its subsequent engine versions.  This cab be a broader discussion on how (and if) we want to have such tests.

To validate this particular change, I've added a sanity test - This test uses hard coded config Json strings to mimic the scenario where users hand-edit the config file instead of using CLI. This config file is used to start the DAB engine and REST and GraphQL requests are executed against it and validated.

Reply:
After discussion in the scrum on 2nd Feb 2024: Need to modify the logic to not write the fields with default values when the option is not used in the command. The older approach has been to write out all the fields, but to avoid bloat in the config file, going forward, we will not write out the fields (ones corresponding to options that are not used) to the config file. This also avoids any issues that could arise when merging two config files.

Reply:
Making the above mentioned update will reduce the size of this pr too hopefully  since no longer updating all the verify.txt files.

Reply:
Discussed in scrum the importance of not writing out to file the default behavior if the property is null

Reply:
I've updated the logic to not write out fields with default values when the option is not used in the command